src/context/AppProvider/AppContext.tsx

```diff
@@ -93,19 +93,18 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useEffect(() => {
     if (!wallet) return
     const walletSupportedChainIds = knownChainIds.filter(chainId => {
-      const chainAccountIds = accountIdsByChainId[chainId] ?? []
-      return walletSupportsChain({ chainId, wallet, isSnapInstalled, chainAccountIds })
+      return walletDeviceSupportsChain({ chainId, wallet, isSnapInstalled })
     })
     dispatch(portfolio.actions.setWalletSupportedChainIds(walletSupportedChainIds))
   }, [accountIdsByChainId, dispatch, isSnapInstalled, wallet])
 
+  // Initial account and portfolio fetch for non-ledger wallets
   useEffect(() => {
-    if (!wallet) return
+    // Skip if wallet is ledger, or if wallet is already connected to accounts - prevents this overriding user selection in account management
+    if (!wallet || isLedger(wallet) || requestedAccountIds.length > 0) return
```

Comment:
This change is key to not overriding changes set in account management. The code following this check is now used on first boot only  for a given device and sets the default "auto-detected" accounts which the user has as the initial basis for managing accounts.

Reply:
Not sure about this one either, see https://github.com/shapeshift/web/pull/7029/files#r1621975666 - we should keep the previous wallet chain support utils as-is (meaning, keep runtime checks as they were previously) instead of adding Ledger-specific logic in here.

The result would (or at least, should) effectively be the same, as the new flavour early returns if if Ledger, and the previous would return the currently connected chains for Ledger. If the previous was ever a problem, then we have bigger problems in our flow.

Reply:
This hook is intended to auto-fetch all accounts up to the first empty one - a feature specific to "non-ledger" wallets, which makes sense to have in plain view here since this rule isn't actually used anywhere else. If we intend of having ledger specific rules elsewhere that requires embedding them into the util lets discuss 🤝

Reply:
> If the previous was ever a problem, then we have bigger problems in our flow.

Correct, this was an issue because it meant we were blowing away the user selections for connected accounts on each boot